apps/worker/src/request_processor/dsnp.announcement.processor.ts

```diff
@@ -0,0 +1,383 @@
+    }
+  }
+
+  public async prepareNote(noteContent?: any): Promise<[string, string, string]> {
+    this.logger.debug(`Preparing note`);
+    const tags: ActivityContentTag[] = this.prepareTags(noteContent?.content.tag);
+    const attachments: ActivityContentAttachment[] = await this.prepareAttachments(noteContent?.content.assets);
+
+    const note = createNote(noteContent?.content.content ?? '', new Date(noteContent?.content.published ?? ''), {
+      name: noteContent?.content.name,
+      location: this.prepareLocation(noteContent?.content.location),
+      tag: tags,
+      attachment: attachments,
+    });
+    const noteString = JSON.stringify(note);
+    const [cid, hash] = await this.pinBufferToIPFS(Buffer.from(noteString));
+    const ipfsUrl = await this.formIpfsUrl(cid);
+    return [cid, hash, ipfsUrl];
+  }
+
+  private prepareTags(tagData?: any[]): ActivityContentTag[] {
+    this.logger.debug(`Preparing tags`);
+    const tags: ActivityContentTag[] = [];
+    if (tagData) {
+      tagData.forEach((tag) => {
+        switch (tag.type) {
+          case TagTypeDto.Hashtag:
+            tags.push({ name: tag.name });
+            break;
+          case TagTypeDto.Mention:
+            tags.push({
+              name: tag.name,
+              type: 'Mention',
+              id: tag.mentionedId,
+            });
+            break;
+          default:
+            throw new Error(`Unsupported tag type ${typeof tag.type}`);
+        }
+      });
+    }
+    return tags;
+  }
+
+  private async prepareAttachments(assetData?: any[]): Promise<ActivityContentAttachment[]> {
+    const attachments: ActivityContentAttachment[] = [];
+    if (assetData) {
+      assetData.forEach(async (asset) => {
+        switch (asset.type) {
+          case AttachmentTypeDto.LINK:
+            attachments.push(this.prepareLinkAttachment(asset));
+            break;
+          case AttachmentTypeDto.IMAGE:
+            attachments.push(await this.prepareImageAttachment(asset));
+            break;
+          case AttachmentTypeDto.VIDEO:
+            attachments.push(await this.prepareVideoAttachment(asset));
+            break;
+          case AttachmentTypeDto.AUDIO:
+            attachments.push(await this.prepareAudioAttachment(asset));
+            break;
+          default:
+            throw new Error(`Unsupported attachment type ${typeof asset.type}`);
+        }
+      });
+    }
+
+    return attachments;
+  }
+
+  private prepareLinkAttachment(asset: AssetDto): ActivityContentLink {
+    this.logger.debug(`Preparing link attachment`);
+    return {
+      type: 'Link',
+      href: asset.href || '',
+      name: asset.name,
+    };
+  }
+
+  private async prepareImageAttachment(asset: AssetDto): Promise<ActivityContentImage> {
+    const imageLinks: ActivityContentImageLink[] = [];
+    asset.references?.forEach(async (reference) => {
+      const assetMetaData = await this.assetQueue.getJob(reference.referenceId);
```

Comment:
So each job should be independent from other jobs. We should not expect this to resolve. This is not a blocker and going to address it in another PR.

Reply:
Yeah i as a bit confused about it, but lets fix it in next